### PR TITLE
mysql-test-run: mention --force in --max-test-fail description

### PR DESCRIPTION
**Pull request by FooBarrior**

### CODE REVIEW

**Comment by grooverdan** [comment]
well highlighted. I use this all the time, prefer the behaviour changed (but have been too slack to get this accepted). Documenting is a good start however so thank you.

File: mysql-test/mysql-test-run.pl [hunk 1]
```diff
@@ -6155,7 +6155,7 @@ ($)
                         $opt_max_save_datadir, set to 0 for no limit. Set
                         it's default with MTR_MAX_SAVE_DATADIR
   max-test-fail         Limit the number of test failures before aborting
-                        the current test run. Defaults to
+                        the current test run. Requires --force. Defaults to
```
Comment (midenok) [inline mysql-test/mysql-test-run.pl]: I'd prefer a straight fix:
```diff
--- a/mysql-test/mysql-test-run.pl
+++ b/mysql-test/mysql-test-run.pl
@@ -1396,6 +1396,11 @@ sub command_line_setup {
     $opt_big_test= 1;
   }
 
+  if ( !$opt_max_test_fail && !$opt_force )
+  {
+    $opt_force= 1;
+  }
+
   # --------------------------------------------------------------------------
   # Find out type of logging that are being used
   # --------------------------------------------------------------------------
@@ -6342,7 +6347,7 @@ Options for debugging the product
                         $opt_max_save_datadir, set to 0 for no limit. Set
                         it's default with MTR_MAX_SAVE_DATADIR
   max-test-fail         Limit the number of test failures before aborting
-                        the current test run. Defaults to
+                        the current test run. Implies --force. Defaults to
                         $opt_max_test_fail, set to 0 for no limit. Set
                         it's default with MTR_MAX_TEST_FAIL
   core-in-failure      Generate a core even if run server is run with valgrind
```

Comment (FooBarrior) [reply mysql-test/mysql-test-run.pl]: Then it makes --force redundant. Maybe delete it completely? Or mark deprecated at least

Comment (midenok) [reply mysql-test/mysql-test-run.pl]: https://github.com/MariaDB/server/pull/1332/commits/fffefd3ce1508e0abc7d01bd59d4b56cb6828dbf

Comment (an3l) [reply mysql-test/mysql-test-run.pl]: So we can treat this patch (fffefd3) as a special case for all tests (`=0`) and is covered by documentation (in patch). 
However `--force` has to be applied while specifying number of failed tests `./mtr --suite=anel --max-test-fail=2 --force` which I see as non-unique rule. At least documentation has to say about this case.

Comment (midenok) [reply mysql-test/mysql-test-run.pl]: Actually `$opt_max_test_fail != 1` is a better condition.

Comment (midenok) [reply mysql-test/mysql-test-run.pl]: > Then it makes --force redundant. Maybe delete it completely? Or mark deprecated at least
I don't know why no `--force` is needed neither. Seems like implementation conflict (that who did second option was unaware of first).

Comment (FooBarrior) [reply mysql-test/mysql-test-run.pl]: https://github.com/MariaDB/server/commit/b2c5019c8b25983cd896b74a62bfdced6c0a578c :-)

Comment (midenok) [reply mysql-test/mysql-test-run.pl]: I would be happy. But maybe @montywi  would not...

**Comment by an3l** [comment]
@FooBarrior , @midenok 
I would prefer to update the PR by applying probably the patch b2c5019 or fffefd3. This will reflect to name of PR and corresponding new jira task.

**Comment by midenok** [comment]
Probably 99c8c691b3d185 will do as discussed with @vuvova. This PR should be closed.

**Pull request closed without merging**